server/controllers/finance/purchases.js

```diff
@@ -288,7 +289,20 @@ function search(req, res, next) {
  * filter the purchase orders.
  */
 function find(options) {
-  const filters = new FilterParser(options, { tableAlias : 'p' });
+  // ensure epected options are parsed appropriately as binary
+  db.convert(options, ['supplier_uuid']);
+
+  const filters = new FilterParser(options, { tableAlias : 'p', autoParseStatements : false  });
+
+  // default purchase date
+  filters.period('period', 'date');
+  filters.dateFrom('custion_period_start', 'date');
+  filters.dateTo('custom_period_end', 'date');
+  filters.equals('user_id');
+  filters.equals('is_confirmed');
```

Comment:
Since you are using checkboxes, you would expect this to make an SQL like:
```sql
is_confirmed = ? OR is_received = ? OR is_canceled = ?
```
That way at least one matches when you search for both being checked.